### PR TITLE
Ban use of print macros in non-test code

### PR DESCRIPTION
**Pull request by wfchandler**
With #746 we added the `_nopipe` variants of the `(e)print(ln)!` macros to avoid panicking when piping to head(1). However, we do not systematically enforce their use, which will inevitably lead inconsistent usage within the project.

Add `clippy` lints to ban use of the print macros in all non-test code. Either the `_nopipe` variants should be used when we don't care if the input is read, or `write!` when the information is crucial, such as an interactive session.

### CODE REVIEW

**Comment by wfchandler** [comment]
For the purposes of `oxide.rs` this PR isn't necessary, the change in https://github.com/oxidecomputer/oxide.rs/pull/751 is enough to sort all subcommands. However, I think this is still a useful default for anyone else using `progenitor`.

**Comment by ahl** [comment]
This is what the rust big-brains came up with?

**Comment by wfchandler** [comment]
> This is what the rust big-brains came up with?

No, I was just looking at what `clippy` could do for us before going to the council of elders and saw these lints.

**Comment by ahl** [comment]
> > This is what the rust big-brains came up with?
> 
> No, I was just looking at what `clippy` could do for us before going to the council of elders and saw these lints.

Worth asking the council of elders? The `_nopipe` suffix strikes me as inelegant.

**Comment by ahl** [comment]
Given https://github.com/oxidecomputer/oxide.rs/pull/774#issuecomment-2272135166 should we consider other approaches?

**Comment by wfchandler** [comment]
I brought this to the [council of big brains going on vacation](https://matrix.to/#/!uMMSVUNAgMdoTtyJPR:oxide.computer/$VYtjP_IeQK1m8PTi7IFmRp8ddiPT6lDWGzxaa8UJQmA?via=oxide.computer&via=unix.house&via=matrix.org). There are a few possibilities, but given our need to support Windows I think the approach in this PR is best.

1. Reset the `SIGPIPE` handler using the [sigpipe](https://crates.io/crates/sigpipe) crate. Ripgrep was [doing this at one point](https://github.com/BurntSushi/ripgrep/pull/586/files), and we've done this in [oxlog](https://github.com/oxidecomputer/omicron/pull/5358), but on [Illumos using Rust < 1.82](https://github.com/oxidecomputer/helios/issues/148) this is not handled properly. More importantly, [this does not work on non-Unix systems](https://github.com/kurtbuilds/sigpipe/blob/master/src/lib.rs#L9-L11).
2. Manually update the panic handler to ignore errors containing the string 'broken pipe'. This is the approach taken by [uutils](https://github.com/uutils/coreutils/blob/a55d9b359d8de34251218fd6f26d433688160556/src/uucore/src/lib/mods/panic.rs#L19-L46), but the consensus was that this is a bit janky.
3. Don't use the standard `print` macros at all; the approach taken in this PR. If you feel that the `_nopipe` macros are too wordy, we could rename them to shadow the `std` macros. The clippy bug is a bit of a pain, but we're aware of it now and I think it's reasonable to watch for it until the lint is fixed.

**Comment by ahl** [comment]
Sounds like options 3 is the best option; and let's stay vigilant in code review given the clippy limitations.

**Review by ahl — APPROVED** [review]
